modules/cql/src/blaze/elm/compiler/type_operators.clj

```diff
@@ -228,7 +228,65 @@
     (->DescendentsOperatorExpression source)))
 
 
-;; TODO 22.18. Is
+;; 22.18. Is
+(defrecord IsExpression [operand type pred]
+  core/Expression
+  (-eval [_ context resource scope]
+    (pred (core/-eval operand context resource scope)))
+  (-form [_]
+    `(~'is ~type ~(core/-form operand))))
+
+
+(defn- matches-elm-named-type-is [type-name]
+  (case type-name
+    "Boolean" ['elm/boolean boolean?]
+    "Integer" ['elm/integer int?]
+    "Long" ['elm/long system/long?]
+    "Decimal" ['elm/decimal decimal?]
+    "Date" ['elm/date system/date?]
+    "DateTime" ['elm/date-time date-time/temporal?]
+    "Quantity" ['elm/quantity quantity/quantity?]
+    "String" ['elm/string string?]))
+
+
+(defn- matches-named-type-is [type-name]
+  (let [[type-ns type-name] (elm-util/parse-qualified-name type-name)]
+    (case type-ns
+      "http://hl7.org/fhir"
+      [(symbol "fhir" type-name)
+       (let [fhir-type (keyword "fhir" type-name)]
```

Comment:
I had to change the match function here, because the set returns either the keyword if it matches or nil, but we need boolean here.